### PR TITLE
add more trace info

### PR DESCRIPTION
**Pull request by tpolecat**
This adds a bit more tracing information so we can see the current user and a bit more information about internal services. Also fixes a cookie-related bug (see below).

### CODE REVIEW

File: modules/service/src/main/scala/CookieService.scala [hunk 1]
```diff
@@ -95,7 +95,7 @@ object CookieWriter {
           name     = CookieName,
           domain   = Some(domain),
           content  = "",
-          sameSite = if (secure) SameSite.Strict else SameSite.Lax,
```
Comment (tpolecat) [inline modules/service/src/main/scala/CookieService.scala:98]: `SameSite.Lax` actually requires https so it doesn't work locally. All very confusing. In any case this fixes the problem and you can run SSO locally with this change.

File: modules/service/src/main/scala/config/Config.scala [hunk 2]
```diff
@@ -85,7 +85,7 @@ object Config {
       keyPair.getPublic,
       keyPair.getPrivate,
       8080,
-      "lucuma.xyz",
+      "local.lucuma.xyz",
```
Comment (tpolecat) [inline modules/service/src/main/scala/config/Config.scala:88]: This is only for local development, which assumes `localhost` is aliased to `local.lucuma.xyz`.

File: modules/service/src/main/scala/database/Database.scala [hunk 3]
```diff
@@ -68,215 +69,262 @@ trait Database[F[_]] {
...
+        Trace[F].span("createGuestUserSessionToken") {
+          s.prepare(InsertGuestUserSessionToken).use(_.unique(guestUser.id))
+        }
 
       def createGuestUserAndSessionToken: F[(GuestUser, SessionToken)] =
-        s.transaction.use { _ =>
-          createGuestUser.mproduct(createGuestUserSessionToken) // 🔥
+        Trace[F].span("createGuestUserAndSessionToken") {
+          s.transaction.use { _ =>
+            createGuestUser.mproduct(createGuestUserSessionToken) // 🔥
```
Comment (toddburnside) [inline modules/service/src/main/scala/database/Database.scala:103]: Just out of curiosity, what is the significance of the fire emoji comment?

**Review by toddburnside — APPROVED** [review]
LGTM

File: modules/service/src/main/scala/database/Database.scala (hunk 3 above)
Comment (tpolecat) [reply modules/service/src/main/scala/database/Database.scala:103]: Functor has `fproduct : F[A] => (A => B) => F[(A, B)]` which is pretty well known. Lesser know is monad's `mproduct: M[A] => (A => M[B]) => M[(A, B)]` so I was excited I had a change to use it.

Comment (toddburnside) [reply modules/service/src/main/scala/database/Database.scala:103]: Nice. 😄

**Pull request merged**